lib/board.rb

```diff
@@ -0,0 +1,86 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells=[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells=[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} \n\n"
+  end
+
+  def position(spot)
+    cells[spot.to_i-1]
+  end
+
+  def update(spot, player)
+      cells[spot.to_i-1] = player.token
+  end
+
+  def all_X
+    ex = []
+    cells.each_with_index do |p, i| 
+      if p == "X"
+        ex << i
+      end
+    end
+    ex
+  end
+
+  def all_O
+    oh = []
+    cells.each_with_index do |p, i| 
+      if p == "O"
+        oh << i
+      end
+    end
+    oh
+  end
+
+  def all_open
+    open = []
+    cells.each_with_index do |p, i| 
+      if p == " "
+        open << i
+      end
+    end
+    open
+  end
+
+  def token
+  end
+
+  def turn_count
+    counter = 0
+    cells.each do |spot|
+      if spot == "X" || spot == "O"
+      counter += 1
+      end
+    end
+    counter
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|spot| spot == "X" || spot == "O"}
end
```